test/livebook_web/live/session_live_test.exs

```diff
@@ -1171,6 +1171,199 @@ defmodule LivebookWeb.SessionLiveTest do
+
+      assert view
+             |> element(
+               ~s{form[phx-submit="create_pvc"] select[name="pvc[storage_class]"] option[value="second-storage-class"]}
+             )
+             |> has_element?()
+
+      assert view
+             |> element(~s{form[phx-submit="create_pvc"] button[type="submit"][disabled]})
+             |> has_element?()
+
+      view
+      |> element(~s{form[phx-submit="create_pvc"]})
+      |> render_change(%{pvc: %{name: "new-pvc", size_gb: 1}})
+
+      assert view
+             |> element(~s{form[phx-submit="create_pvc"] button[type="submit"]:not([disabled])})
+             |> has_element?()
+
+      Req.Test.expect(Livebook.Runtime.K8s, Livebook.ClusterStub)
+
+      view
+      |> element(~s{form[phx-submit="create_pvc"]})
+      |> render_submit(%{pvc: %{name: "new-pvc", size_gb: 1}})
+
+      Req.Test.verify!()
+
+      # Delete a PVC
+
+      view
+      |> element(~s{button[phx-click="delete_pvc"]})
+      |> render_click()
+
+      assert render_async(view) =~
+               "Are you sure you want to irreversibly delete Persistent Volume Claim"
+
+      Req.Test.expect(Livebook.Runtime.K8s, Livebook.ClusterStub)
+
+      view
+      |> element(~s{button[phx-click="confirm_delete_pvc"]})
+      |> render_click()
+
+      Req.Test.verify!()
+
+      # Pod Template Validation
+
+      refute render_async(view) =~ ~s/Make sure to define a valid resource of apiVersion /
+
+      view
+      |> element(~s{form[phx-change="set_pod_template"]})
+      |> render_change(%{pod_template: ""})
+
+      assert render_async(view) =~ ~s/Make sure to define a valid resource of apiVersion /
+
+      view
+      |> element(~s{form[phx-change="set_pod_template"]})
+      |> render_change(%{
+        pod_template: """
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          generateName: livebook-runtime-
+        spec:
+          containers:
+            - name: other-name
+        """
+      })
+
+      assert render_async(view) =~ ~s/Main container is missing./
+
+      # We do not actually connect the runtime. We test connecting againast the
+      # real API separately
+    end
+
+    test "populates k8s runtime config form existing runtime", %{conn: conn, session: session} do
+      pod_template = """
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        generateName: livebook-runtime-
+        labels:
+          livebook.dev/component: test
+      spec:
+        containers:
+          - name: livebook-runtime\
+      """
+
+      runtime =
+        Runtime.K8s.new(
+          %{
+            context: "default",
+            namespace: "default",
+            home_pvc: "foo-pvc",
+            docker_tag: "nightly",
+            pod_template: pod_template
+          },
+          nil
+        )
+
+      Req.Test.stub(Livebook.Runtime.K8s, Livebook.ClusterStub)
+      Req.Test.stub(Livebook.Runtime.K8s.NoPermissionCluster, Livebook.NoPermissionClusterStub)
```

Comment:
```suggestion
```

Reply:
Hmm... if I remove this, I get:

```
     ** (EXIT from #PID<0.1004.0>) an exception was raised:
         ** (RuntimeError) cannot find mock/stub Livebook.Runtime.K8s.NoPermissionCluster in process #PID<0.1011.0>
             (req 0.5.2) lib/req/test.ex:388: Req.Test.__fetch_plug__/1
             (req 0.5.2) lib/req/test.ex:621: Req.Test.call/2
             (req 0.5.2) lib/req/steps.ex:894: Req.Steps.run_plug/1
             (req 0.5.2) lib/req/request.ex:1101: Req.Request.run_request/1
             (req 0.5.2) lib/req/request.ex:1045: Req.Request.run/1
             (livebook 0.15.0-dev) lib/livebook/k8s/auth.ex:51: Livebook.K8s.Auth.create_self_subject_access_review/2
             (elixir 1.16.2) lib/task/supervised.ex:101: Task.Supervised.invoke_mfa/2
             (elixir 1.16.2) lib/task/supervised.ex:36: Task.Supervised.reply/4
```

Reply:
Interesting, at a glance I though we don't interact with the no-permission cluster in this test.

Reply:
Or is the no-permission picked as the default? Then I think the error would be solved by moving `set_context` from mount to update as per another comment.

Reply:
oooh that's it! Nice catch!